esp/smc/SMCLib/TpWrapper.cpp

```diff
@@ -464,8 +464,52 @@ void CTpWrapper::getTpEspServers(IArrayOf<IConstTpEspServer>& list)
     }
 }
 
+static Owned<IEspTpMachine> createLocalHostTpMachine(const char *path)
+{
+    Owned<IEspTpMachine> machine = createTpMachine();
+    IpAddress ipAddr;
+    ipAddr.ipset("localhost");
+    StringBuffer localHost;
+    ipAddr.getIpText(localHost);
+    machine->setName(localHost.str());
+    machine->setNetaddress(localHost.str());
+    machine->setConfigNetaddress("localhost");
+    machine->setDirectory(path);
+    machine->setOS(getPathSepChar(path) == '/' ? MachineOsLinux : MachineOsW2K);
+    return machine.getClear();
+}
+
 void CTpWrapper::getTpDfuServers(IArrayOf<IConstTpDfuServer>& list)
 {
+#ifdef _CONTAINERIZED
+    Owned<IPropertyTreeIterator> dfuQueues = queryComponentConfig().getElements("dfuQueues");
+    ForEach(*dfuQueues)
+    {
+        IPropertyTree & dfuQueue = dfuQueues->query();
+        const char * dfuName = dfuQueue.queryProp("@name");
+        StringBuffer queue;
+        getDfuQueueName(queue, dfuName);
+        Owned<IEspTpDfuServer> pService = createTpDfuServer("","");
+        pService->setName(dfuName);
+        pService->setDescription(dfuName);
+        pService->setBuild("");
+        pService->setQueue(queue);
+        pService->setType(eqDfu);
+        pService->setLogDirectory("/var/log/HPCCSystems/dfuserver");
+        IArrayOf<IEspTpMachine> tpMachines;
+        Owned<IPropertyTreeIterator> planes = getDropZonePlanesIterator();
+        ForEach(*planes)
+        {
+            IPropertyTree & plane = planes->query();
+            const char * path = plane.queryProp("@prefix");
+
+            if (!isEmptyString(path))
+                tpMachines.append(*createLocalHostTpMachine(path).getClear());
```

Comment:
createLocalHostTpMachine should just return a regular pointer that is owned by caller semantically